packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -146,23 +119,23 @@ ToolbarLink.displayName = LINK_NAME;
 
 const TOGGLE_GROUP_NAME = 'ToolbarToggleGroup';
 
-type ToolbarToggleGroupOwnProps = Polymorphic.OwnProps<typeof ToggleGroupPrimitive.Root>;
-type ToolbarToggleGroupPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof ToggleGroupPrimitive.Root>,
-  ToolbarToggleGroupOwnProps
->;
-
-const ToolbarToggleGroup = React.forwardRef((props, forwardedRef) => {
-  const context = useToolbarContext(TOGGLE_GROUP_NAME);
-  return (
-    <ToggleGroupPrimitive.Root
-      data-orientation={context.orientation}
-      {...props}
-      ref={forwardedRef}
-      rovingFocus={false}
-    />
-  );
-}) as ToolbarToggleGroupPrimitive;
+type ToolbarToggleGroupElement = React.ElementRef<typeof ToggleGroupPrimitive.Root>;
+type ToolbarToggleGroupProps = React.ComponentPropsWithoutRef<typeof ToggleGroupPrimitive.Root>;
+
+const ToolbarToggleGroup = React.forwardRef<ToolbarToggleGroupElement, ToolbarToggleGroupProps>(
+  (props, forwardedRef) => {
+    const context = useToolbarContext(TOGGLE_GROUP_NAME);
+    return (
+      // @ts-ignore
+      <ToggleGroupPrimitive.Root
```

Comment:
In the next PR I'll be exporting all of our component types so that we no longer need to do `React.ComponentPropsWithoutRef<typeof ...>` because that utility flattens union which breaks [discriminated unions](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions).

In the meantime, I've ignored this so we can start reviewing this work separately from that piece.